pkg/api/v1beta1/defaults.go

```diff
@@ -93,7 +96,42 @@ func init() {
 		},
 		func(obj *Endpoints) {
 			if obj.Protocol == "" {
-				obj.Protocol = "TCP"
+				obj.Protocol = ProtocolTCP
+			}
+			if len(obj.Subsets) == 0 && len(obj.Endpoints) > 0 {
```

Comment:
If they are both non-empty we should probably force the legacy fields to
match the subsets.  I think I just let some corner cases go here since they
are unlikely to happen (I think)

On Thu, Mar 26, 2015 at 10:21 PM, Brian Grant <notifications@github.com>
wrote:

> In pkg/api/v1beta1/defaults.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5939#discussion_r27276557>
> :
>
> > @@ -93,7 +96,42 @@ func init() {
> >  		},
> >  		func(obj *Endpoints) {
> >  			if obj.Protocol == "" {
> > -				obj.Protocol = "TCP"
> > +				obj.Protocol = ProtocolTCP
> > +			}
> > +			if len(obj.Subsets) == 0 && len(obj.Endpoints) > 0 {
>
> What should happen if both are nonempty?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5939/files#r27276557>
> .
>